FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataWriter/Azure/AzureBlobContainerClient.cs

```diff
@@ -428,5 +445,10 @@ public async IAsyncEnumerable<PathItem> ListPathsAsync(string directory, [Enumer
             }
             while (item != null);
         }
+
+        private static bool IsAuthenticationError(RequestFailedException exception) =>
+            string.Equals(exception.ErrorCode, AzureStorageErrorCode.NoAuthenticationInformationErrorCode, StringComparison.OrdinalIgnoreCase) ||
```

Comment:
Shall we check only 403 code from [doc](https://learn.microsoft.com/en-us/rest/api/storageservices/common-rest-api-error-codes)